cmd/juju/controller/killstatus.go

```diff
@@ -99,6 +103,13 @@ func newData(api destroyControllerAPI, controllerModelUUID string) (ctrData, []m
 	var aliveModelCount int
 	var ctrModelData modelData
 	for _, model := range status {
+		if newerAPI && model.Error != nil {
+			// This most likely occurred because a model was
+			// destroyed half-way through the call.
+			// Since we filter out models with life.Dead below, it's safe
```

Comment:
It's not safe though, because we don't *know* that it's dead. Feels like too big of an assumption to me. It would be fine to check for "NotFound", I think. For other errors, can we wait and try again?

Reply:
Added a check for NotFound, as discussed. All other errors will be bubbled up.